src/app/examples/crud/components/crud.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, NgForm } from '@angular/forms';
+
+import { takeUntil, map } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+
+import { ROUTE_ANIMATIONS_ELEMENTS } from '@app/core';
+
+import { State } from '../../examples.state';
+import { Book } from '../books.model';
+import {
+  ActionBooksPersist,
+  AddOne,
+  DeleteOne,
+  UpdateOne,
+  ActionBookSelect
+} from '../books.actions';
+import {
+  selectBooks,
+  selectCurrentBook,
+  selectAllBooks
+} from '../books.selectors';
+
+@Component({
+  selector: 'anms-crud',
+  templateUrl: './crud.component.html',
+  styleUrls: ['./crud.component.scss']
+})
+export class CrudComponent implements OnInit {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+
+  routeAnimationsElements = ROUTE_ANIMATIONS_ELEMENTS;
+
+  emptyBook: Book = {
+    id: new Date().getUTCMilliseconds().toString(),
+    title: '',
+    author: '',
+    description: ''
+  };
+  bookFormGroup = this.fb.group(this.emptyBook);
+  books: Book[];
+  selectedBook: Book | null;
+  isEditing: boolean;
+
+  constructor(public store: Store<State>, public fb: FormBuilder) {}
+
+  ngOnInit() {
+    this.store
```

Comment:
Would it make sense to handle the persistence inside effects, instead of the component itself?

Reply:
Yes it could be cleaner but i'm searching a way to achieve it. 
I'm trying something with the effect bellow but without success.
```typescript
@Effect({dispatch: false})
  persistBooks() {
    return this.actions$.pipe(
      ofType<AddOne|UpdateOne|DeleteOne>(
        BookActionTypes.ADD_ONE,
        BookActionTypes.UPDATE_ONE,
        BookActionTypes.DELETE_ONE),
      tap(_ => {
        this.store.select(state => {
          return this.localStorageService.setItem(
            BOOKS_KEY,
            state.examples.books
          );
        });
      })
    );
```
@timdeschryver or @tomastrajan do you have an idea?

Reply:
I think instead of using `select` in the `tap` operator, you could use `withLatestFrom(this.store.pipe(select(selector)))`.

If this isn't the case I suppose we can solve it in a separate issue, because I think this is the way how it's currently done in every example.

Reply:
see the following article, find `4. Using a selector inside your effects`:

https://blog.angularindepth.com/start-using-ngrx-effects-for-this-e0b2bd9da165

Reply:
OK thanks for trying 😄 , let's do this in an other issue and in every example.